src/main/java/net/minecraftforge/client/extensions/IForgePoseStack.java

```diff
@@ -0,0 +1,42 @@
+/*
+ * Copyright (c) Forge Development LLC and contributors
+ * SPDX-License-Identifier: LGPL-2.1-only
+ */
+
+package net.minecraftforge.client.extensions;
+
+import com.mojang.blaze3d.vertex.PoseStack;
+import com.mojang.math.Transformation;
+import com.mojang.math.Vector3f;
+
+/**
+ * Extension interface for {@link com.mojang.blaze3d.vertex.PoseStack}.
+ */
+public interface IForgePoseStack
+{
+    private PoseStack self()
```

Comment:
Just a note: This is not really a good idea in anything rendering related.
The JVM hates these private interface methods with a cast, cause it has to do a double check on if this interface implementation is actually the PoseStack. In other words we know this, but the Runtime does not.

We should really find a better way of doing this.....
But for now this is fine.

Reply:
The other alternative is to define the method inside of `PoseStack` directly via patch, either keeping the extension interface for the method definition + javadocs and using `{@inheritDoc}`, or forgoing the interface and keeping it all in the patch.

Reply:
Also possible, the problem is the `(PoseStack) this` statement in the interface. As long as it is not in the interface the problem is solved.